src/mono/wasm/Wasm.Build.Tests/NativeLibraryTests.cs

```diff
@@ -136,5 +136,36 @@ public static int Main()
             string cryptoInitMsg = "MONO_WASM: Initializing Crypto WebWorker";
             Assert.DoesNotContain(cryptoInitMsg, output);
         }
+
+        [Theory]
+        [BuildAndRun(aot: false)]
+        [BuildAndRun(aot: true)]
+        public void ProjectWithNativeLibrary(BuildArgs buildArgs, RunHost host, string id)
+        {
+            string projectName = $"AppUsingNativeLibrary-a";
+            buildArgs = buildArgs with { ProjectName = projectName };
+            buildArgs = ExpandBuildArgs(buildArgs, extraItems: "<NativeLibrary Include=\"native-lib.o\" />");
```

Comment:
```suggestion
            buildArgs = ExpandBuildArgs(buildArgs, extraItems: "<NativeLibrary Include=\"native-lib.o\" />\n<NativeLibrary Include=\"DoesNotExist.o\" \>");
```

Reply:
This will fail the build, right? It would be ideal to have it conditional as theory data

Reply:
We need to make it not fail the build since NativeLibrary does not mind if the file doesn't exist